src/nbl/video/CVulkanDescriptorSet.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __NBL_VIDEO_C_VULKAN_DESCRIPTOR_SET_H_INCLUDED__
+
+#include "nbl/video/IGPUDescriptorSet.h"
+
+#include <volk.h>
+
+namespace nbl::video
+{
+
+class ILogicalDevice;
+class CVulkanDescriptorPool;
+
+class CVulkanDescriptorSet : public IGPUDescriptorSet
+{
+public:
+    CVulkanDescriptorSet(core::smart_refctd_ptr<ILogicalDevice>&& dev,
+        core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& layout,
+        core::smart_refctd_ptr<const CVulkanDescriptorPool> parentPool, VkDescriptorSet descriptorSet)
+        : IGPUDescriptorSet(std::move(dev), std::move(layout)), m_parentPool(parentPool),
+        m_descriptorSet(descriptorSet)
+    {}
+
+    inline VkDescriptorSet getInternalObject() const { return m_descriptorSet; }
+
+private:
+    core::smart_refctd_ptr<const CVulkanDescriptorPool> m_parentPool = nullptr;
```

Comment:
not const, will need it

Reply:
the descriptor set should do some of what OpenGL descriptor set does, which is keep track of the `IReferenceCounted` objects which are written to it

For this I suggest:
1. working out the count of objects from the descset layout
2. allocate an appropriate sized slab from the descriptor pool parent
3. using the layout to map <set,binding,index> tuples to linear offsets into the slab
4. add a TODO note to make sure the slab can be sparse/segmented so that one can use descriptor indexing with partially populated descriptor sets